packages/frontend/components/Access/GuildAccess.tsx

```diff
@@ -0,0 +1,188 @@
+import { Check, Close, LogoutOutlined } from '@mui/icons-material';
+import LoadingButton from '@mui/lab/LoadingButton';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  IconButton,
+  Paper,
+  Stack,
+  Tooltip,
+  Typography,
+  useMediaQuery,
+} from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import Image from 'next/image';
+import { useEffect, useState } from 'react';
+
+import { AccessStatus, useAccess } from '../../store/access.store';
+import { AuthStatus, useAuth } from '../../store/auth.store';
+
+export function GuildAccess() {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
+
+  const [isOpen, setIsOpen] = useState(true);
+
+  const address = useAuth((state) => state.address);
+  const authStatus = useAuth((state) => state.status);
+  const login = useAuth((state) => state.login);
+  const logout = useAuth((state) => state.logout);
+
+  const accessStatus = useAccess((state) => state.status);
+  const retriesCount = useAccess((state) => state.retriesCount);
+  const errorsCount = useAccess((state) => state.errorsCount);
+  const verify = useAccess((state) => state.verify);
+  const reset = useAccess((state) => state.reset);
+
+  useEffect(() => {
+    if (address) verify();
+  }, [address, verify]);
+
+  useEffect(() => {
+    if (accessStatus === AccessStatus.Verifying) return;
+
+    if (accessStatus === AccessStatus.Verified) {
+      // close this modal 3 secs after the verification succeeded
+      setTimeout(() => {
+        setIsOpen(false);
+      }, 3000);
+    } else {
+      setIsOpen(true);
+    }
+  }, [accessStatus]);
+
+  const statusParagraph = () => {
+    if (authStatus !== AuthStatus.Connected) return <></>;
+
+    if (accessStatus === AccessStatus.Verifying) {
```

Comment:
better to use content table for this aka
const statusLabels = { AccessStatus.Verifying: 'Verifying your eligibility ...',  }
and get string like this statusLabels[accessStatus]